geps/gep-1742.md

```diff
@@ -301,13 +301,168 @@ Could not find any HTTP specific timeouts. PRs welcomed. 😊
 
 ## API
 
-TBD.
+The above diagrams show that there are many different kinds of configurable timeouts
+supported by Gateway implementations: connect, idle, request, upstream, downstream.
+Although there may be opportunity for the specification of a common API for more of
+them in the future, this GEP will focus on the L7 timeouts in HTTPRoutes that are
+most valuable to clients.
+
+From the above analysis, it appears that, at a minimum, implementations are capable of
+supporting the configuration of simple client downstream request timeouts on HTTPRoute
+rules. This is a relatively small addition that would benefit many users.
+
+Some implementations support configuring a timeout for individual backend requests,
+separate from the overall client request timeout. This is particularly useful if a
+client HTTP request to a gateway can result in more than one call from the gateway
+to the destination backend service, for example, if automatic retries are supported.
+Adding non-Core support for this would also benefit many users.
+
+### GO
+
+```go
+type HTTPRouteRule struct {
+	// Timeouts defines the timeouts that can be configured for an HTTP request.
+	//
+	// Support: Core
+	//
+	// +optional
+	// <gateway:experimental>
+	Timeouts *HTTPRouteTimeouts `json:"timeouts,omitempty"`
+
+	// ...
+}
+
+// HTTPRouteTimeouts defines timeouts that can be configured for an HTTPRoute.
+// Timeout values are formatted like 1h/1m/1s/1ms as parsed by Golang time.ParseDuration
+// and MUST BE >= 1ms.
+type HTTPRouteTimeouts struct {
+	// Request specifies the duration for processing an HTTP client request after which the
+	// gateway will time out if unable to send a response.
+	// Whether the gateway starts the timeout before or after the entire client request stream
+	// has been received, is implementation-dependent.
```

Comment:
I agree that it would be better to not have this implementation dependent start time. The problem is that (based on the evaluation that @youngnick did) we believe that it can't be supported by **all** implementations. So, the alternative is to make the feature "Extended", but we instead are trying to use this compromise so that we can at least have something be part of "Core" support. FYI, here is some of the background discussion on this: https://github.com/kubernetes-sigs/gateway-api/pull/1997#discussion_r1195375173

Reply:
I interpret that discussion a little differently than saying it is implementation-dependent.  It doesn't look like there was a conclusion on it, but I think @youngnick is asking for a better clarification on what exactly we're measuring here.  It's still just a little too vague.

Reply:
@candita, I added more detailed doc in the comments of both timeout fields. PTAL.